src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+    eventTooltip.reset();
+    gridRowSizes = gridRowSizesNoBottomPanel;
+    selectedSource = null;
+  }
+
+  function deselectEvent() {
+    selectedEvent = null;
+    selectedRowId = null;
+  }
+
+  function onCanvasMouseDown(e: CustomEvent<MouseDown>) {
+    if (!mouseDownAfterTable) {
+      const { externalEvents } = e.detail;
+      // selectedEvent is our source of an ExternalEvent as well as the ExternalEventId used by this instance
+      //    of the LayerExternalSources (as opposed to using a store, like the timeline one does, which is
+      //    unnecessary as everything we need is in on single component or can be passed down via parameters to
+      //    children).
+      selectedEvent = externalEvents?.length ? externalEvents[0] : null;
+      selectedRowId = null;
+    } else {
+      mouseDownAfterTable = false;
+    }
+  }
+
+  function onCanvasMouseOver(e: CustomEvent<MouseOver>) {
+    mouseOver = e.detail; // Allows tooltip to access object
+  }
+
+  function toggleItem(value: ExternalSourceType) {
+    if (!selectedFilters.find(f => f.name === value.name)) {
+      selectedFilters = selectedFilters.concat(value);
+    } else {
+      selectedFilters = selectedFilters.filter(filter => filter.name !== value.name);
+    }
+  }
+
+  function unselectFilteredValues() {
+    selectedFilters = [];
+  }
+
+  function selectFilteredValues() {
+    selectedFilters = [...new Set([...selectedFilters, ...filteredValues])];
+  }
+
+  function onSelectionChanged() {
+    selectedEvent = selectedEvents.find(event => getRowIdExternalEvent(event.pkey) === selectedRowId) ?? null;
+  }
+
+  function onManageGroupsAndTypes() {
+    effects.manageGroupsAndTypes(user);
+  }
+</script>
+
+<CssGrid columns={uiColumnSize}>
+  <Panel borderRight padBody={true}>
+    <svelte:fragment slot="header">
+      <SectionTitle
+        >{selectedEvent
+          ? `Selected Event (${selectedEvent.pkey.key})`
+          : selectedSource
+            ? `Selected External Source (${selectedSource.pkey.key})`
+            : 'Upload a Source File'}</SectionTitle
+      >
+      {#if selectedEvent}
+        <button
+          class="st-button icon fs-6"
+          on:click={deselectEvent}
+          use:tooltip={{ content: 'Deselect event', placement: 'top' }}
+        >
+          <XIcon />
+        </button>
+      {:else if selectedSource}
+        <button
+          class="st-button icon fs-6"
+          on:click={deselectSource}
+          use:tooltip={{ content: 'Deselect source', placement: 'top' }}
+        >
+          <XIcon />
+        </button>
+      {/if}
+    </svelte:fragment>
+
+    <svelte:fragment slot="body">
+      {#if selectedEvent}
+        <ExternalEventForm externalEvent={selectedEvent} showHeader={true} />
+      {:else if selectedSource}
+        <div class="external-source-header">
+          <div class={classNames('external-source-header-title')}>
+            <div class="external-source-header-title-value st-typography-medium">
```

Comment:
Could this be a disabled input instead of a div? I know our disabled inputs are not ideal for this but we do have plans to update how we handle read-only form elements. Would be good to be consistent with how we approach this in other forms.
<img width="385" alt="image" src="https://github.com/user-attachments/assets/62f1ddc7-0aeb-4240-92b4-893e746a9139">

Reply:
This panel (or at least the one for selected events) was modeled after the selected activity panel! We thought mirroring that and separating out the name in its own title would be the most readable.

Reply:
Resolved.